controllers/openstackdataplanenode_controller.go

```diff
@@ -166,3 +187,99 @@ func (r *OpenStackDataPlaneNodeReconciler) ConfigureNetwork(ctx context.Context,
+	if instance.Spec.Node.AnsiblePort > 0 {
+		return strconv.Itoa(instance.Spec.Node.AnsiblePort)
+	}
+	return strconv.Itoa(instanceRole.Spec.NodeTemplate.AnsiblePort)
+}
+
+// GetAnsibleManaged retruns the string (from boolean) value from the template unless it is set in the node
+func (r *OpenStackDataPlaneNodeReconciler) GetAnsibleManaged(instance *dataplanev1beta1.OpenStackDataPlaneNode, instanceRole *dataplanev1beta1.OpenStackDataPlaneRole) string {
+	if instance.Spec.Node.Managed {
+		return strconv.FormatBool(instance.Spec.Node.Managed)
+	}
+	return strconv.FormatBool(instanceRole.Spec.NodeTemplate.Managed)
+}
+
+// GetAnsibleManagementNetwork retruns the string value from the template unless it is set in the node
+func (r *OpenStackDataPlaneNodeReconciler) GetAnsibleManagementNetwork(instance *dataplanev1beta1.OpenStackDataPlaneNode, instanceRole *dataplanev1beta1.OpenStackDataPlaneRole) string {
+	if instance.Spec.Node.ManagementNetwork != "" {
+		return instance.Spec.Node.ManagementNetwork
+	}
+	return instanceRole.Spec.NodeTemplate.ManagementNetwork
+}
+
+// GetAnsibleNetworkConfig retruns a JSON string value from the template unless it is set in the node
+func (r *OpenStackDataPlaneNodeReconciler) GetAnsibleNetworkConfig(instance *dataplanev1beta1.OpenStackDataPlaneNode, instanceRole *dataplanev1beta1.OpenStackDataPlaneRole) string {
+	if instance.Spec.Node.NetworkConfig != instanceRole.Spec.NodeTemplate.NetworkConfig {
+		return fmt.Sprintf("{template: %s}", instance.Spec.Node.NetworkConfig)
+
+	}
+	return fmt.Sprintf("{template: %s}", instanceRole.Spec.NodeTemplate.NetworkConfig.Template)
+}
+
+// GetAnsibleNetworks retruns a JSON string mapping fixedIP and/or network name to their valules
+func (r *OpenStackDataPlaneNodeReconciler) GetAnsibleNetworks(instance *dataplanev1beta1.OpenStackDataPlaneNode, instanceRole *dataplanev1beta1.OpenStackDataPlaneRole) string {
+	var network string // the resulting string containing each network
+	for _, netMap := range instance.Spec.Node.Networks {
+		if netMap.FixedIP != "" && netMap.Network == "" {
+			network += fmt.Sprintf("{%s: %s},", "fixedIP", netMap.FixedIP)
+		}
+		if netMap.FixedIP != "" && netMap.Network != "" {
+			network += fmt.Sprintf("{%s: %s, %s: %s},",
+				"fixedIP", netMap.FixedIP, "network", netMap.Network)
+		}
+	}
+	return fmt.Sprintf("[%s]", strings.TrimSuffix(network, ","))
+}
+
+// EnsureNodeOnRoleList will ensure that a node is on its role's dataPlaneNodes list
+func (r *OpenStackDataPlaneNodeReconciler) EnsureNodeOnRoleList(ctx context.Context, instance *dataplanev1beta1.OpenStackDataPlaneNode, instanceRole *dataplanev1beta1.OpenStackDataPlaneRole) error {
+	// 1. When a node is created by a role it will already be on the list
+	// 2. When a node is directly created with a role name, it won't be on the list
+	// In the second case we have to add the node directly and set nodeFrom field
+
+	var err error
+	var nodeOnList bool // default is false
```

Comment:
you can  codify that comment  with `nodeOnList := false` and that is even shorter :)

Reply:
Thanks, I've done that in my code and it will be reflected in the next patch set that I push.